workflow/common/configmap.go

```diff
@@ -4,14 +4,17 @@ import (
 	"fmt"
 
 	apiv1 "k8s.io/api/core/v1"
-	"k8s.io/client-go/tools/cache"
 
 	"github.com/argoproj/argo-workflows/v3/errors"
 )
 
+type ConfigMapStore interface {
```

Comment:
I use a simpler interface for utility functions for testability, otherwise we need a mock informer for test cases, if this is inappropriate, please leave a comment, thanks!

Reply:
good idea

Reply:
Thanks, any further advice?